packages/protocol/RepaintGraphicsCache.ts

```diff
@@ -11,6 +12,20 @@ export const RepaintGraphicsCache: Cache<
   debugLabel: "RepaintGraphicsCache",
   getKey: ([replayClient, pauseId]) => pauseId,
   load: async ([replayClient, pauseId]) => {
-    return replayClient.repaintGraphics(pauseId);
+    const result = await replayClient.repaintGraphics(pauseId);
+
+    let { description, screenShot } = result;
+
+    // The backend won't return a screenshot for a given hash more than once; see FE-2357
+    if (screenShot) {
+      paintHashCache.cacheValue(screenShot, description.hash);
+    } else {
+      screenShot = paintHashCache.getValueIfCached(description.hash);
+    }
```

Comment:
[`DOM.repaintGraphics` API docs](https://static.replay.io/protocol/tot/DOM/#method-repaintGraphics) say this about the `screenShot` value:
> Contents of the screen shot. If an identical screen shot was previously returned, this will be omitted.